jupyterlab/datastore/handler.py

```diff
@@ -96,150 +85,130 @@ def get(self, *args, **kwargs):
         yield super(WSBaseHandler, self).get(*args, **kwargs)
 
     def get_compression_options(self):
-        return self.settings.get('websocket_compression_options', None)
+        return self.settings.get("websocket_compression_options", None)
 
 
 class CollaborationHandler(WSBaseHandler):
     """Request handler for the datastore API"""
 
-    def initialize(self):
+    def initialize(self) -> None:
         self.log.info("Initializing datastore connection %s", self.request.path)
-        self.collaboration = None
+        self.collaboration: typing.Optional[Collaboration] = None
         self.collaboration_id = None
-        self.store_id = None
         self.store_transaction_serial = -1
         self.history_inited = False
 
     @property
     def datastore_file(self):
-        return self.settings.setdefault('datastore_file', ':memory:')
+        return self.settings.setdefault("datastore_file", ":memory:")
 
     @property
     def rtc_recovery_timeout(self):
-        return self.settings.get('rtc_recovery_timeout', 120)
+        return self.settings.get("rtc_recovery_timeout", 120)
 
-    @gen.coroutine
-    def pre_get(self):
-        # authenticate first
-        super(CollaborationHandler, self).pre_get()
-
-        self.store_id = self.get_query_argument('storeId', None)
-        # TODO: Check if store id supplied, if so, check if already open
-        if self.store_id is not None:
-            # TODO: Temporary minimal reaction:
-            self.log.warning("Trying to reopen store that is already connected")
-            raise web.HTTPError(400)
```

Comment:
The bit before this acts as a pretty decent placeholder (raises if trying to connect multiple times with the same store id).

Reply:
We should probably also fail here if the `store_id` is not given.

Reply:
I removed the backend knowledge of the `store_id` altogether. I actually think this is  maybe ok? It just identifies users by connection.  If you lose connection, then you make a new one and it treats you as a new peer.

Reply:
It possibly makes things like checkpoints a little harder. I'll make a separate issue for checkpoints. We can reintroduce this concept if needed after that.